src/libs/ui/src/molecules/FilterDropdown/FilterDropdown.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useRef, useEffect, ReactNode } from 'react';
+
+interface FilterDropdownProps {
+  options: string[];
+  onSelect: (option: string) => void;
+  children?: ReactNode; // Accept children
+}
+
+const FilterDropdown: React.FC<FilterDropdownProps> = ({ options, onSelect, children }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedOption, setSelectedOption] = useState(options[0]);
+  const [prevOption, setPrevOption] = useState(options[0]);
+  const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+  const toggleDropdown = () => {
+    setIsOpen((prev) => !prev);
+  };
+
+  const handleOptionSelect = (option: string) => {
+    setPrevOption(selectedOption); // Save previous option
+    setSelectedOption(option);
+    onSelect(option); // Send selected option to parent
+    setIsOpen(false); // Close dropdown after selection
+  };
+
+  // Close the dropdown when clicking outside of it
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
+        setIsOpen(false);
+        setSelectedOption(prevOption); // Revert to previous option if closed
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [prevOption]);
+
+  return (
+    <div className="relative inline-block" ref={dropdownRef}>
+      <button
+        onClick={toggleDropdown}
+        className={`flex items-center px-4 py-2 bg-white border rounded-full shadow-sm focus:outline-none ${isOpen ? 'border-blue-500' : 'border-gray-300'}`}
+      >
+        <svg
```

Comment:
Please read Svg from assets Folder

Reply:
Moved the SVG icon to the assets folder.
![image](https://github.com/user-attachments/assets/e8ac6389-523f-48dd-9a97-1a6e021f0f9e)